Feature/estimate cox

AlexandraP-21 opened the pull request:
**What changes are proposed in this pull request?**

**Which files are involved in this pull request and what changes were made?**

**Did you include unit tests for the proposed change/bug fix (https://testthat.r-lib.org/)?**

**If there is an GitHub issue associated with this pull request, please provide link.**

--------------------------------------------------------------------------------

Checklist for PR reviewer

- [ ] PR branch has pulled the most recent updates from main branch. Ensure the pull request branch and your local version match and both have the latest updates from the main branch.
- [x] If a new function was added, function should be included in `_pkgdown.yml`
- [x] If a bug was fixed, a unit test was added for the bug check
- [x] Run `pkgdown::build_site()`. Check the R console for errors, and review the rendered website.
- [x] Code coverage is suitable for any new functions/features. Review coverage with `withr::with_envvar(new = c("NOT_CRAN" = "true"), covr::report())`. Before you run, begin a fresh R session without any packages loaded. 
- [ ] R CMD Check runs without errors, warnings, and notes
- [ ] `usethis::use_spell_check()` runs with no spelling errors in documentation
- [x] Has `NEWS.md` been updated with the changes from this pull request under the heading indicating the latest version. If there is an issue associated with the pull request, reference it in parentheses at the end update (see `NEWS.md` for examples).
- [ ] Has the version number been incremented using `usethis::use_version(which = "dev")` 
- [ ] Approve Pull Request
- [ ] Merge the PR. Please use "Squash and merge".

AlexandraP-21 committed:
1. reexports: survival::tt gives an error. Needs further investigation.
1i. estimate_cox: Works without calling 'library(survival)'!

AlexandraP-21 committed:
1. Create extimate_cox and testing

AlexandraP-21 committed:
1. Added library(survival) to check if it works now

AlexandraP-21 committed:
1. Take out library(survival)

ddsjoberg committed:
redocumented

AlexandraP-21 committed:
1. Added testing for `conf.int` argument

AlexandraP-21 committed:
1. Remove glue

AlexandraP-21 committed:
Merge branch 'develop' into feature/estimate_cox

SHAESEN2 commented on tests/testthat/test-estimate_cox.R:
```diff
@@ -0,0 +1,358 @@
+#' T3.2 No error when equation is NULL
+#' T3.3 When no equation is specified, the 'null model' is displayed
+#' T3.4 When the equation has a term with only one factor, expect an error
+#' T4.  The function removes all rows with NA values inside CNSR or AVAL
```
T4 should test conf.int see estimate_cuminc and below should be upgraded by 1 eg T4 becomes T5

AlexandraP-21 replied:
Done

SHAESEN2 commented on tests/testthat/test-estimate_cox.R:
```diff
@@ -0,0 +1,358 @@
+#' T4.1 The function removes all rows with NA values inside CNSR or AVAL
+#' T5. The function does not alter the calculation of survival::coxph
+#' T5.1 The function gives the same results as survival::coxph
+#' T5.2 The function allows `conf.int` argument to be passed, specific for summary(survival::coxph)
```
should be T4.

AlexandraP-21 replied:
Changed numbering

SHAESEN2 commented on tests/testthat/test-estimate_cox.R:
```diff
@@ -0,0 +1,358 @@
+#' T6.1 The calculation is not affected by the addition of additional parameters
+#' T6.2 The function add PARAM/PARAMCD when available
+#' T7. The function call supports traceability
+#' T7.1 The function updates call$data when magrittr pipe is used
```
call["data"] => also update estimate_KM if I wrote that there

AlexandraP-21 replied:
Done in both.

SHAESEN2 commented on tests/testthat/test-estimate_cox.R:
```diff
@@ -0,0 +1,358 @@
+#' T5.2 The function allows `conf.int` argument to be passed, specific for summary(survival::coxph)
+#' T5.3 The function allows additional arguments to be passed, specific for survival::coxph & summary(survival::coxph)
+#' T5.4 The function returns an object of class `coxph`
+#' T6. The function adds additional information to the survfit object when available
```
what about the summary information? also needs to be tested

AlexandraP-21 replied:
Arent we already checking it with the 'conf.int' and the additional components of the summary.coxph?

actions-user committed:
Re-build README.Rmd

AlexandraP-21 committed:
Merge branch 'develop' into feature/estimate_cox

AlexandraP-21 commented on R/estimate_cox.R:
```diff
@@ -0,0 +1,173 @@
+#'    The function expects that the data has been filtered on the parameter (PARAM/PARAMCD) of interest.
+#'    The result is an object of class \code{coxph} which can be used in downstream functions and methods that rely on the \code{coxph} class.
+#'
+#' @param data The name of the dataset for the Cox model is based on the Analysis Data Model (ADaM) principles. The dataset is expected to have
```
Added @inheritParams survival::coxph

AlexandraP-21 replied:
and @inheritParams survival::summary.coxph

AlexandraP-21 commented on R/estimate_cox.R:
```diff
@@ -0,0 +1,173 @@
+#' @param conf.int Confidence intervals used for the estimation.
+#' @param ... additional arguments passed on to the ellipsis of the call \code{survival::coxph(data = data, formula = survival::Surv(AVAL, 1-CNSR) ~ equation), ...)} .
+#'    Use \code{?survival::coxph} and \code{?survival::summary.coxph} for more information.
+#'
```
I see it there

AlexandraP-21 commented on R/estimate_cox.R:
```diff
@@ -0,0 +1,173 @@
+#' @title Wrapper for Cox Proportional Hazard analysis
+#'
+#' @description This function is a wrapper around \code{survival::coxph} to fit a Cox Proportional Hazards Regression model, assuming right-censored data.
+#'    The function expects that the data has been filtered on the parameter (PARAM/PARAMCD) of interest.
```
We could, but 'ties' and other arguments fall in the '...' and are not seen as params of the function. Should I add it as another param in the function and thus in the documentation?

SHAESEN2 replied:
I agree with you :)

bailliem replied:
Should we add a badge to indicate this function is experimental?

i.e. add in the description block 
#' `r lifecycle::badge("experimental")`

AlexandraP-21 commented on R/estimate_cox.R:
```diff
@@ -0,0 +1,173 @@
+#'    Default is NULL.
+#' @param AVAL Analysis value for the Cox model. Default is "AVAL", as per CDISC ADaM guiding principles.
+#' @param CNSR Censor for the Cox model. Default is "CNSR", as per CDISC ADaM guiding principles.
+#' @param conf.int Confidence intervals used for the estimation.
```
Added the following: #' @param conf.int Confidence intervals used for the estimation. Conf.int argument is used from the \code{survival::summary.coxph}. The default is 0.95.

AlexandraP-21 commented on R/estimate_cox.R:
```diff
@@ -0,0 +1,173 @@
+  
+  summary_cox <- summary(survfit_object, conf.int = conf.int)
+  
+  survfit_object[["conf.int"]]      <- summary_cox[["conf.int"]]
```
You have something specific in mind?

SHAESEN2 replied:
I see now that the information also changes eg coefficients. Your approach is OK for now.

AlexandraP-21 commented on R/estimate_cox.R:
```diff
@@ -0,0 +1,173 @@
+#' @param ... additional arguments passed on to the ellipsis of the call \code{survival::coxph(data = data, formula = survival::Surv(AVAL, 1-CNSR) ~ equation), ...)} .
+#'    Use \code{?survival::coxph} and \code{?survival::summary.coxph} for more information.
+#'
+#' @return a coxph model
```
Added:
 #' @return an object of coxph, with additional components of the class of\code{survival::summary.coxph}. Use \code{?survival::coxph} and \code{?survival::summary.coxph} for more information.

AlexandraP-21 commented:
**Issue for discussion:** 

- `coxph` returns the components 'coefficients' and 'concordance'
- `summary.coxph` returns the components 'coefficients' and 'concordance'

These are (kind of) different.

**_Example:_**
fit1 <- coxph(data = adtte, 
              Surv( AVAL,  1- CNSR ) ~ AGE + strata(TRTA) )
sumfit <-summary(fit1)

#-------------------------------------------------------------------------------------#

**fit1$coefficients**
        AGE 
-0.01462011 

**sumfit$coefficients**
                 coef             exp(coef)       se(coef)                z               Pr(>|z|)
AGE   -0.01462011     0.9854862     0.009421986    -1.551702    0.1207336

**fit1$concordance**
  concordant         discordant               tied.x              tied.y                 tied.xy                 concordance          std 
3.816000e+03   3.175000e+03    2.640000e+02     6.700000e+01    1.000000e+00    5.441764e-01    2.631382e-02 

**sumfit$concordance**
         C      se(C) 
0.54417643 0.02631382

#-------------------------------------------------------------------------------------#
For the moment I changed the naming in the `estimate_cox`, so for the `summary.coxph` it returns:
  survfit_object[["**coef.exp**"]]      <- summary_cox[["coefficients"]]
  survfit_object[["**concor.se**"]]     <- summary_cox[["concordance"]]


Any other ideas on how to handle this are welcome.

AlexandraP-21 committed:
1. Made updates for estimate_cox and the testing according to some comments on GitHub.
2. For the test-estimate_KM I just changed something minor in the documentation (call["data"]  instead of call$data).

AlexandraP-21 committed:
Merge branch 'feature/estimate_cox' of https://github.com/openpharma/visR into feature/estimate_cox

actions-user committed:
Re-build README.Rmd

SHAESEN2 commented:
I would look at what Daniel is doing for estimate_KM to save the environment and the expression and implement something similar in a new branch.

ddsjoberg committed:
Merge branch 'develop' into feature/estimate_cox

actions-user committed:
Re-build README.Rmd

ddsjoberg committed:
Merge commit '0303a0557fb2053141065be06e6634dc36aae283'

actions-user committed:
Re-build README.Rmd

ddsjoberg committed:
Merge branch 'develop' into feature/estimate_cox

bailliem commented:
Merge quosure PR first
Then apply to this PR

bailliem commented on R/estimate_cox.R:
```diff
@@ -0,0 +1,186 @@
+#' # Stratified Kaplan-Meier analysis by `TRTP`
+#' visR::estimate_cox(data = adtte, equation = "TRTP")
+#'
+#' ## Multivariate Cox model using the covariates: `TRTP` and `SEX`
```
Could we change multivariate to multivariable.

bailliem commented on R/estimate_cox.R:
```diff
@@ -0,0 +1,186 @@
+  ,AVAL = "AVAL"
+  ,conf.int = 0.95
+  ,...
+){
```
TBD - could we change equation to formula to be consistent with estimate_KM and general R terminology. I think we could be ok with not displaying the y side when CDISC, but also provides space to then have the complete formula for non CDISC data

ddsjoberg committed:
Merge commit 'c13980f40e344d2ce24b92fa98c84b2c1e4d43c7'

ddsjoberg committed:
merge conflict updates

ddsjoberg committed:
Merge branch 'develop' into feature/estimate_cox

AlexandraP-21 committed:
1. Change 'multivariate' to 'multivariable'
2. Added thats its experimental.
3. Updated NEWS.md
4. Updated _pkgdown.yml

AlexandraP-21 committed:
Increment version number to 0.2.0.9012
